src/argilla/server/schemas/v1/datasets.py

```diff
@@ -287,9 +288,19 @@ class RecordInclude(str, Enum):
     responses = "responses"
 
 
+class RecordGetterDict(GetterDict):
+    def get(self, key: Any, default: Any = None) -> Any:
+        if key == "metadata":
+            return getattr(self._obj, "metadata_", {})
+        if key == "responses" and "responses" not in self._obj.__dict__:
```

Comment:
Now I understand the change in handlers. Much better like this 👍

Reply:
I just realise that we don't need this custom `GetterDict`. I added this to try to make it work with the default `from_orm` method, but if we return `None` in this condition `if key == "responses" and "responses" not in self._obj.__dict__:`, `pydantic` will detect that we have set a value for the `Record.responses` attribute, and then in the listing records endpoints `response_model_exclude_unset=True` won't affect this field, and we will have `"responses: null"` in the output payload.

Reply:
Okay, so we keep using record.__dict___?

Reply:
yes